journal-api/src/main/java/com/liferay/journal/util/JournalStringUtil.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.journal.util;
+
+import com.liferay.portal.kernel.util.CharPool;
+import com.liferay.portal.kernel.util.StringBundler;
+import com.liferay.portal.kernel.util.StringPool;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.util.Stack;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Pavel Savinov
+ */
+public class JournalStringUtil {
+
+	public static String shortenWithHtml(String string, int length) {
```

Comment:
In case the string.length() is less than length requirement, we still end up parsing the string and trying to pairing up the tag. Which is not only wasteful but also wrong.
For example, if the string is simply "<a>", we should just return it, as the length default to 200. But the actual return value is "<a>...</a>", no way this can be right.

Please resend with unit test for this method to show me what the expected logic is.